src/librustc_mir/transform/copy_prop.rs

```diff
@@ -31,13 +31,13 @@ use rustc::ty::TyCtxt;
 pub struct CopyPropagation;
 
 impl<'tcx> MirPass<'tcx> for CopyPropagation {
-    fn run_pass(&self, tcx: TyCtxt<'tcx>, _source: MirSource<'tcx>, body: &mut BodyAndCache<'tcx>) {
+    fn pass_type(&self) -> MirPassType {
         // We only run when the MIR optimization level is > 1.
         // This avoids a slow pass, and messing up debug info.
-        if tcx.sess.opts.debugging_opts.mir_opt_level <= 1 {
-            return;
-        }
+        MirPassType::Optimization(MirOptimizationLevel::Unstable)
```

Comment:
The ">1" and `Unstable` is pretty disconnected now. Might be more transparent to just return the `min_opt_level` at which the optimization runs, or so?